ansible/playbooks/adhoc/hotfix/roles/update_node/tasks/main.yml

```diff
@@ -18,42 +15,41 @@
   changed_when: False
 
 # ansible will return a '17' if a match is found. '-1' means 'not found'
-- name: Set update_successful=True if upgrade succeeded
+- name: Set wanted_version_installed=True if upgrade succeeded
   set_fact: 
-    update_successful: True
+    wanted_version_installed: True
   when: "{{ post_upgrade.stdout.find(version_wanted) != -1 }}"
 
 - name: Debug node vars
-  debug: msg="update_successful={{update_successful}}, rpm_installed={{post_upgrade.stdout}}, version_wanted={{version_wanted}}"
+  debug: msg="wanted_version_installed={{wanted_version_installed}}, rpm_installed={{post_upgrade.stdout}}, version_wanted={{version_wanted}}"
 
 - name: Upgrade failed, so try a downgrade
   command: "yum -y downgrade atomic-openshift-{{ version_wanted }} atomic-openshift-clients-{{ version_wanted }} atomic-openshift-node-{{ version_wanted }} tuned-profiles-atomic-openshift-node-{{ version_wanted }} atomic-openshift-sdn-ovs-{{ version_wanted }}"
-  when: "{{ update_successful == False }}"
+  when: "{{ wanted_version_installed == False }}"
 
 - name: Check if downgrade is successful
   command: "rpm -qa atomic-openshift"
   register: post_downgrade
   changed_when: False
 
 # ansible will return a '17' if a match is found. '-1' means 'not found'
-- name: Set downgrade_successful=True if downgrade succeeded
+- name: Set wanted_version_installed=True if downgrade succeeded
   set_fact: 
-    downgrade_successful: True
-    upgrade_successful: True
+    wanted_version_installed: True
   when: "{{ post_downgrade.stdout.find(version_wanted) != -1 }}"
 
 - name: Alert if update is not successful, but continue
   debug:
     msg: "[WARNING] The desired rpm version does not match installed version! Upgrade and downgrade failed!"
-  when: "{{ update_successful == False and downgrade_successful == False }}"
+  when: "{{ wanted_version_installed == False }}"
   failed_when: True
   ignore_errors: True
 
 - name: Restart atomic-openshift-node service if upgrade was successful
   service:
     name: atomic-openshift-node
     state: restarted
-  when: "{{ update_needed == True and update_successful == True }}" 
+  when: "{{ update_needed == True and wanted_version_installed == True }}"
 
 - name: Ensure node service is running
   service:
```

Comment:
I think `ignore_errors` makes the most sense on the `oadm` command. That command will retry 10 times if it fails initially, so `ignore_errors` allows it to retry without causing the whole playbook to terminate. 

The reason I have so many retry attempts is that, in the past, Preview has been very busy and these calls would fail sometimes. But it's been working a lot better lately.

In the case of the nodes, I thought we might want to continue even if one node-restart fails. But I'm really fine either way. I can remove it, or keep it. What do you think?